### ISSUE TITLE
List[Literal] input selection not working as expected

### ISSUE THREAD

**Issue by j-tr** [PrefectHQ/prefect#10170]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I refreshed the page and this issue still occurred.
- [X] I checked if this issue was specific to the browser I was using by testing with a different browser.

### Bug summary

It is not possible anymore to select individual items in input fields of type List[Literal]. This was possible in previous Prefect versions.

### Reproduction

```
#!/usr/bin/env python3

from typing import List, Literal
from prefect import flow


@flow(name="test_flow")
def main(
    list_of_literals: List[Literal["foo", "bar"]] = ["foo", "bar"],
) -> None:
    pass


if __name__ == "__main__":
    main()
```

### Error

it is not possible to deselect one of the default values.

see screen recording:
https://www.loom.com/share/b00ac5a91d1b4658a0f900bcab973c47?sid=4dd711c2-3882-4275-a222-7622d169ba24

### Browers

- [X] Chrome
- [X] Firefox
- [ ] Safari
- [ ] Edge

### Prefect version

```Text
Version:             2.10.19
API version:         0.8.4
Python version:      3.8.15
Git commit:          c924f424
Built:               Thu, Jul 6, 2023 1:10 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         ephemeral
Server:
  Database:          sqlite
  SQLite version:    3.37.2

Also reproducible in prefect cloud
```

**Comment by znicholasbrown** [comment]
Thanks for the reproduction @j-tr - this looks to be a regression upstream; we'll see what's up 👍🏻

**Comment by pleek91** [comment]
@j-tr this fix will go out with the next release

**Comment by EmilRex** [comment]
@j-tr this has been released to cloud as well as included in version 2.10.20
